include/ebpf_structs.h

```diff
@@ -77,3 +77,31 @@ typedef enum
     BPF_FUNC_get_smp_processor_id = 7,
     BPF_FUNC_ktime_get_ns = 8,
 } ebpf_helper_id_t;
+
+// Libbpf itself requires the following structs to be defined, but doesn't
+// care what fields they have.  Applications such as bpftool on the other
+// hand depend on fields of specific names and types.
+
+struct bpf_link_info
+{
+    ebpf_id_t id;      ///< Link ID.
+    ebpf_id_t prog_id; ///< Program ID.
+};
+
+struct bpf_map_info
```

Comment:
should bpf_map_info also have map name?

Reply:
There's other fields not covered by this PR, but I went ahead and added it now.  It surfaced a bug where map_name was never being set in the kernel which is also now fixed in this PR.